### PR TITLE
Instance: Use exitErr.ExitCode function for containers

### PR DESCRIPTION
**Pull request by tomponline**
This is available from Go 1.12.
Also removes potential panic situation.

Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>

### CODE REVIEW

**Review by brauner — APPROVED** [review]
Where do we take that function from? And is it guaranteed to return correct shell exit code semantics?

**Comment by tomponline** [comment]
> Where do we take that function from? And is it guaranteed to return correct shell exit code semantics?

The ExitError (https://pkg.go.dev/os/exec#ExitError), we were previously getting too, embeds `os.ProcessState` which has an `ExitCode()` function (https://pkg.go.dev/os#ProcessState.ExitCode), which is implemented as https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/os/exec_posix.go;l=131 and uses the System-dependent status info `status` field which is of type `syscall.WaitStatus` (the same one we were using directly before), which then calls `ExitStatus` on the WaitStatus (https://pkg.go.dev/syscall#WaitStatus.ExitStatus) and thats implemented as:

https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/syscall/syscall_linux.go;l=339-344

File: lxd/instance/drivers/driver_lxc_cmd.go [hunk 1]
```diff
@@ -36,20 +35,11 @@ func (c *lxcCmd) Signal(sig unix.Signal) error {
...
-		exitErr, ok := err.(*exec.ExitError)
-		if ok {
-			status, ok := exitErr.Sys().(syscall.WaitStatus)
-			if ok {
-				return status.ExitStatus(), nil
-			}
-
-			if status.Signaled() {
-				// 128 + n == Fatal error signal "n"
-				return 128 + int(status.Signal()), nil
```
Comment (tomponline) [inline lxd/instance/drivers/driver_lxc_cmd.go:48]: @brauner what I'm not clear about was whether we ever ran this code path, as presumably if we called `status.Signaled()` bit on `status` when it wasn't of type `syscall.WaitStatus` we would get a panic. It seems like this was a dead code path.

**Comment by tomponline** [comment]
> Where do we take that function from? And is it guaranteed to return correct shell exit code semantics?

We use this approach a few other places already, such as:

https://github.com/lxc/lxd/blob/2e490215707b4a5b798287b7416d82ec6b99e692/lxd/storage/drivers/driver_ceph_utils.go#L182-L203

**Comment by brauner** [comment]
Can I merge this? Seems a bit red but likely unrelated?

**Comment by tomponline** [comment]
> Can I merge this? Seems a bit red but likely unrelated?

Yes, I can't see any failures on `lxc exec` or similar. Thanks

**Pull request merged**